pipeline_utils/lib/yaml_parser.py

```diff
@@ -0,0 +1,414 @@
+#!/usr/bin/env python
+
+import yaml
+
+
+###############################################################
+#   Functions
+###############################################################
+def load_yaml(file):
+    """
+        return a generator to loaded yaml documents in file
+    """
+    with open(file) as stream:
+        try:
+            for d in yaml.safe_load_all(stream):
+                yield d
+        except yaml.YAMLError as exc:
+            sys.exit(exc)
```

Comment:
Pretty sure you are missing `sys` import here, so another exception will be thrown that will mask the original one?